docs: clarify `BlockIDFlag` variants (#9590)
tendermint/tendermint: spec/core/data_structures.md

a particular `BlockID` or was absent. It's a part of the `Commit` and can be used
to reconstruct the vote set given the validator set.

| Name             | Type                        | Description                                                                                                                                                      | Validation                                                        |
|------------------|-----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------|

> | BlockIDFlag      | [BlockIDFlag](#blockidflag) | Represents the validators participation in consensus: voted for another block or did not vote,, voted for the block that received the majority, or voted for nil | Must be one of the fields in the [BlockIDFlag](#blockidflag) enum |

Reviewer: ```suggestion
| BlockIDFlag      | [BlockIDFlag](#blockidflag) | Represents the validators participation in consensus: its vote was not received, voted for the block that received the majority, or voted for nil | Must be one of the fields in the [BlockIDFlag](#blockidflag) enum |
```

Reviewer: Similar reasoning to my previous comment

```go
enum BlockIDFlag {
  BLOCK_ID_FLAG_UNKNOWN = 0; // indicates an error condition

>   BLOCK_ID_FLAG_ABSENT  = 1; // voted for another block or did not vote

Reviewer: ```suggestion
  BLOCK_ID_FLAG_ABSENT  = 1; // the vote was not received
```
